src/javascripts/services/travel_tracker_service.js

```diff
@@ -0,0 +1,42 @@
+function fetchUsername() {
+  const fetchApi = fetch('https://fe-apps.herokuapp.com/api/v1/travel-tracker/data/travelers/travelers');
+  fetchApi.then(response => {
+    return response.json();
+  }).then(data => {
+    var travArray = (data.travelers);
+
+    // Change username variable to loginInput
+    var loginInputBefore = document.getElementById('input-username');
+    // Downcase loginInput and chop
+    var loginInput = loginInputBefore.value.toLowerCase();
+    // First 9 letters saved to variable called loginName
+    var loginName = loginInput.substring(0, 8);
+    // Right side of split saved to variable called loginId
+    var loginId = loginInput.substring(8, 10);
+
+    // If (loginName === "traveler") then successful username
+    if (loginName === "traveler" || loginName === "agency") {
+      console.log("Username OK")
+    } else {
+      console.log("Username NOT OK")
+    };
+    if (loginId == "0" || loginId == "00") {
+      alert("Please enter a valid Username");
+      document.getElementById("input-username").value = "";
+      document.getElementById("input-password").value = "";
+    }
+    //  If (loginId === an ID in the db) then succesful ID
+    if (loginId <= travArray.length && loginId > 0) {
+      console.log("ID OK")
+    }
+    // If (succesful ID && succesful username) then return user object 
+    if (loginName == "traveler" && loginId <= travArray.length && loginId !== "0") {
+      console.log(travArray[loginId - 1]);
+      alert("Username has been authenticated!")
```

Comment:
All this logic can be moved in to similar helper methods as listed in the login_helper file comments